src/sentry/uptime/models.py

```diff
@@ -228,15 +228,16 @@ def get_audit_log_data(self):
         }
 
 
-def get_org_from_uptime_monitor(uptime_monitor: ProjectUptimeSubscription) -> tuple[Organization]:
-    return (uptime_monitor.project.organization,)
+def get_org_from_detector(detector: Detector) -> tuple[Organization]:
+    return (detector.project.organization,)
 
 
-@cache_func_for_models([(ProjectUptimeSubscription, get_org_from_uptime_monitor)])
+@cache_func_for_models([(Detector, get_org_from_detector)])
 def get_active_auto_monitor_count_for_org(organization: Organization) -> int:
-    return ProjectUptimeSubscription.objects.filter(
+    return Detector.objects.filter(
+        type=UptimeDomainCheckFailure.slug,
         project__organization=organization,
-        mode__in=[
+        config__mode__in=[
             ProjectUptimeSubscriptionMode.AUTO_DETECTED_ONBOARDING,
             ProjectUptimeSubscriptionMode.AUTO_DETECTED_ACTIVE,
         ],
```

Comment:
I didn't know we could query json like this, cool. This part of the query will be slow, but I think it's ok since we're already constrained by type/project.

We actually don't have an index on `(project, type)`, that could be worth adding. Maybe in a separate pr?